biochemist-python/chapters/Nonlinear Curve Fitting Part 1.ipynb

```diff
@@ -0,0 +1,1436 @@
+     "execution_count": 41,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Calculate initial velocities and place those in a new column in the dataframe\n",
+    "MM_df['initial velocities'] = MM_df['slopes'] / 0.015\n",
+    "MM_df"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will use this dataframe now to perform the nonlinear regression fit using the SciPy library in part 2 of this lesson. To save this data for part 2, so we need to write it to a csv file in our data directory."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 42,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "MM_df.to_csv('biochemist-python/chapters/data/MM_data.csv')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"exercise-this-is-a-title exercise\">\n",
+    "<p class=\"exercise-title\">Check your understanding</p>\n",
+    "    <p>You will find an Excel file in your data folder, chymotrypsin_kinetics.xlsx, with some kinetic data from a chymotrypsin experiment. Apply the principles above to create dataframes and a .csv file for creating a Michaelis-Menten plot with these data. Under these assay conditions the extinction coefficient for p-nitrophenol is 18,320 M<sup>-1</sup>cm<sup>-1</sup>.</p>\n",
+    "\n",
+    "```{admonition} Hint\n",
+    ":class: dropdown\n",
+    "    You will need to get the data into a layout and file format that is easily read by pandas. \n",
+    "    <ul>\n",
+    "        <li>Delete the first seven lines of the Excel file.</li>\n",
+    "        <li>Delete the first column of the Excel file.</li>\n",
+    "        <li>Save the file as chymotrypsin_kinetics.csv.</li>\n",
+    "        <li>Your data will should look something like this:</li>\n",
+    "        <img src=\"biochemist-python/chapters/images/csv_image.png\" alt=\"csv image\">\n",
+    "    \n",
+    "```{admonition} Solution\n",
+    ":class: dropdown\n",
+    "    \n",
+    "```python\n",
+    "        import os \n",
+    "        import pandas as pd \n",
+    "        import numpy as np \n",
+    "        from scipy import stats \n",
+    "        datafile = os.path.join('biochemist-python', 'chapters', 'data', 'chymotrypsin_kinetics.csv') # filehandle created\n",
+    "        chymo_rates_df = pd.read_csv(datafile)\n",
+    "        chymo_rates_df = chymo_rates_df.set_index('Time (sec)')\n",
+    "        \n",
+    "        def linregress_slope(df_series):\n",
+    "            slope, _, _, _, _, = stats.linregress(df_series.index, df_series.values)\n",
+    "            return slope\n",
+    "\n",
+    "        chymo_MM_df = pd.DataFrame()\n",
+    "        chymo_MM_df['slopes'] = chymo_rates_df.apply(linregress_slope)\n",
+    "        chymo_MM_df['Initial Velocities'] = chymo_MM_df['slopes'] / 0.01832\n",
+    "        chymo_MM_df.to_csv('biochemist-python/chapters/data/chymo_MM_data.csv')\n",
```

Comment:
If we use `os.path.join` to build a file path for opening files, we should also use it for writing files.

Reply:
I added os.path.join for writing the csv files for the module and for the take home exercise. I used this approach. 

outputfile = os.path.join('data', 'chymo_MM_data.csv')
chymo_MM_df.to_csv(outputfile)

is there a better way? Can this be done with one line of code?